### ISSUE TITLE
defaultValue does not work with input when type is set to submit

### ISSUE THREAD

**Issue by jakubkosinski** [facebook/react#14785]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

When you use uncontrolled `<input type="submit" />` and set `defaultValue` attribute, it would be ignored in versions 1.5.0 or higher (there would be no `value` attribute in the HTML result). It was working correctly in older versions. Looks like only `type="submit"` is affected, for other input types `defaultValue` behaves correctly.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**

Correct behaviour with react 16.4.2: https://codepen.io/anon/pen/zePmrZ
Incorrect behaviour with react 16.8.1: https://codepen.io/anon/pen/PVOyqV

**What is the expected behavior?**

When `defaultValue="foo"` is set on `<input type="submit"/>` it should result in `<input type="submit" value="foo" />` in the HTML result.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

All versions starting from 16.5.0

**Comment by augbog** [comment]
Someone please correct me if I'm wrong but this might have been intentional? I was curious so stepping through the code, it looks like it bails out from setting the initial `value` in `ReactDOMInput#postMountWrapper` 

```
    // Avoid setting value attribute on submit/reset inputs as it overrides the
    // default value provided by the browser. See: #12872
    if (isButton && (props.value === undefined || props.value === null)) {
      return;
    }
```
Looking at the #12872, it seems the reason for this is that the browser itself has a `defaultValue` for these types (i.e. `submit` and `reset`). I guess this is a question of if these types should allow you to set a `defaultValue`?

**Comment by leonascimento** [comment]
@augbog I agree with you, today tracking this possible regression or bug, I arrived in this [piece of code](https://github.com/facebook/react/blob/c954efa70f44a44be9c33c60c57f87bea6f40a10/packages/react-dom/src/client/ReactDOMInput.js#L251) too.

I'm thinking on this possible fix: when the `input` is a button, why not check if exists the `_initialValue` and assign it to `node.value`?

```jsx
 if (_initialValue) {
   node.value = _initialValue;
 }
```

Something like it:
```jsx
    var _initialValue = toString(node._wrapperState.initialValue);                                                                                                      
                                                                                                                                                                        
    // Avoid setting value attribute on submit/reset inputs as it overrides the                                                                                         
    // default value provided by the browser. See: #12872                                                                                                               
    if (isButton && (props.value === undefined || props.value === null)) {                                                                                              
      if (_initialValue) {                                                                                                                                              
        node.value = _initialValue;                                                                                                                                     
      }                                                                                                                                                                 
      return;                                                                                                                                                           
    } 
```

I did this local fix and is working here.
If you everyone agree with it, could I to work on it?
